keps/sig-cloud-provider/20190423-instance-host-labelling.md

```diff
@@ -0,0 +1,266 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Many kubernetes clusters are deployed on virtual machines (also referred to as
+"instances"), and information about the physical host for those instances (also
+referred to as "instance host") is not consistently available.  Instances which
+share a host can fail simultaneously if there is a problem with the host.  We
+propose to label nodes with their instance host ID so that controllers (namely,
+the scheduler) can decrease shared fate of pods from the same collection.
+
+## Motivation
+
+This KEP is motivated by the desire to improve reliability of workloads running
+in kubernetes clusters.  Today, the scheduler will attempt to spread pods from a
+given collection (e.g. a deployment) across nodes.  Even if every pod ends up on
+a different node, there may be multiple pods per instance host. This means that
+the deployment is not as reliable as it appears to be.
+
+By adding an instance host label, we can treat instance hosts as a failure
+domain, and perform spreading appropriately.
+
+
+### Goals
+
+- Introduce a well known label for instance host (e.g.
+  `failure-domain.kubernetes.io/instance-host`)
+- Add a cloud provider API to get instance host ID
+- Update cloud-controller-manager to update node labels based on the cloud
+  provider value.
+
+### Non-Goals
+
+- Scheduling behavior that considers instance host
+- Controller behavior (e.g. deployment, statefulset) that considers instance
+  host
+- Any decision-making that is based on the instance host label
+
+## Proposal
+
+We will add a new field to the [Zone struct][zone-struct], `PhysicalMachineID`.
+If supported, it will be populated by cloud providers whenever returning a zone.
+
+[zone-struct]: https://github.com/kubernetes/cloud-provider/blob/0a4f4cbb5a664deb4639d7d9bf5bbde3bb3603c1/cloud.go#L208-L211
+
+When the `PhysicalMachineID` is supplied, cloud-controller-manager will use it
+as the value for the `failure-domain.kubernetes.io/physical-host` label.
```

Comment:
Why "failure-domain" when the others have moved to "topology" ?

Reply:
To be fair, we haven't moved to `failure-domain` yet :P but we should use `topology.kubernetes.io` assuming https://github.com/kubernetes/enhancements/pull/839 is 👍

Reply:
Intention is to match the others. It's a moving target. (: